src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3149,28 +3149,34 @@ static char* map_or_reserve_memory_aligned(size_t size, size_t alignment, int fi
   assert(extra_size >= size, "overflow, size is too large to allow alignment");
 
   char* aligned_base = NULL;
+  static const int max_attempts = 20;
```

Comment:
Stylistic review as well: I'd put that counter in a loop like `for (int attempts = 0; aligned_base == NULL && attempts < 20 /* max attemps */; attemps++) { ... }`. That would clearly indicate that it is a counted loop.

Otherwise, now that you can return `NULL` from this function, have you checked that all uses of this function deal with a NULL-returned value correctly? From a cursory glance, it looks good to me.